.github/workflows/build-and-test.yml

```diff
@@ -784,6 +775,86 @@ jobs:
         if: ${{ failure() && runner.debug }}
         uses: canonical/action-tmate@main
 
+  juju-remote-relation-test:
+    needs:
+      - lint
+      - unit-test
+      - build
+    name: Remote Integration test
+    runs-on: self-hosted-linux-amd64-noble-xlarge
+    steps:
+      - name: Download charm
+        uses: actions/download-artifact@v4
+        with:
+          name: charms
+          path: ~/artifacts/
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup LXD
+        uses: canonical/setup-lxd@v0.1.1
+        with:
+          # pin lxd to LTS release.
+          channel: 5.21/stable
+
+      - name: Install dependencies
+        run: ./tests/scripts/ci_helpers.sh install_deps
+
+      - name: Install MicroCeph charm
+        run: |
+          set -eux
+          date
+          # Deploy 2 microceph clusters named primary and secondary
+          juju deploy ~/artifacts/microceph.charm primary --storage osd-standalone='2G,3' --constraints="virt-type=virtual-machine root-disk=50G mem=8G"
+          juju deploy ~/artifacts/microceph.charm secondary --storage osd-standalone='2G,3' --constraints="virt-type=virtual-machine root-disk=50G mem=8G"
+          # wait for charm to bootstrap and OSD devices to enroll.
+          juju wait-for unit primary/0 --timeout '30m' --query='workload-message=="(workload) charm is ready"'
+          juju wait-for unit secondary/0 --timeout '15m' --query='workload-message=="(workload) charm is ready"'
+          bash ./tests/scripts/ci_helpers.sh check_osd_count primary/0 3
+          bash ./tests/scripts/ci_helpers.sh check_osd_count secondary/0 3
+          date
+
+      - name: Add remote integration
+        run: |
+          set -eux
+          juju integrate primary:remote-provider secondary:remote-requirer
+
+          # wait for the applications to go to blocked state
+          juju wait-for unit primary/0 --query='workload-status=="blocked"' --timeout=10m
+          juju wait-for unit secondary/0 --query='workload-status=="blocked"' --timeout=10m
+
+          juju config primary site-name="primary"
+          juju config secondary site-name="secondary"
+
+
+          juju wait-for unit primary/0 --query='workload-status=="active"' --timeout=10m
+          juju wait-for unit secondary/0 --query='workload-status=="active"' --timeout=10m
```

Comment:
Nits: suggest to add `juju status --relations` for troubleshooting; +get rid of the extra blank line

Reply:
ack